src/rules/strictBooleanExpressionsRule.ts

```diff
@@ -0,0 +1,116 @@
+        // then check is the next expression is of type boolean
+        let isExclamationOperator = node.operator === ts.SyntaxKind.ExclamationToken;
+        if (isExclamationOperator) {
+            let expr = node.operand;
+            let expType = checker.getTypeAtLocation(expr);
+            if (!this.isBooleanType(expType)) {
+                this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.UNARY_EXPRESSION_ERROR));
+            }
+        }
+        super.visitPrefixUnaryExpression(node);
+    }
+
+    public visitIfStatement(node: ts.IfStatement): void {
+        let checker = this.getTypeChecker();
+        // check if the expression is of boolean type
+        let expType = checker.getTypeAtLocation(node.expression);
+        if (!this.isBooleanType(expType)) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.IF_STATEMENT_ERROR));
+        }
+        super.visitIfStatement(node);
+    }
+
+    public visitWhileStatement(node: ts.WhileStatement): void {
+        let checker = this.getTypeChecker();
+        // check if the expression is of boolean type
+        let expType = checker.getTypeAtLocation(node.expression);
+        if (!this.isBooleanType(expType)) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.WHILE_STATEMENT_ERROR));
+        }
+        super.visitWhileStatement(node);
+    }
+
+    public visitDoStatement(node: ts.DoStatement): void {
+        let checker = this.getTypeChecker();
+        // check if the expression is of boolean type
+        let expType = checker.getTypeAtLocation(node.expression);
+        if (!this.isBooleanType(expType)) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.DO_STATEMENT_ERROR));
+        }
+        super.visitDoStatement(node);
+    }
+
+    public visitConditionalExpression(node: ts.ConditionalExpression): void {
+        let checker = this.getTypeChecker();
+        // check if the expression is of boolean type
+        let expType = checker.getTypeAtLocation(node.condition);
+        if (!this.isBooleanType(expType)) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.CONDITIONAL_EXPRESSION_ERROR));
+        }
+        super.visitConditionalExpression(node);
+    }
+
+    public visitForStatement(node: ts.ForStatement): void {
+        let checker = this.getTypeChecker();
+        // check if the expression is of boolean type
+        let expType = checker.getTypeAtLocation(node.condition);
+        if (!this.isBooleanType(expType)) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FOR_STATEMENT_ERROR));
+        }
+        super.visitForStatement(node);
+    }
+
+    private isBooleanType(btype: ts.Type): boolean {
+        let boolType = /* tslint:disable:no-bitwise */(btype.flags & ts.TypeFlags.BooleanLike)/* tslint:enable:no-bitwise */;
```

Comment:
@nchen63 this would not work because I am checking the type flags and not the node flags, there are two options available
1. I can leave this line as is
2. I can introduce a new function `isTypeFlagsSet` into `language/utils.ts` which can be used to check the type flags here
which option do you suggest I go with?